Table column initial width setting

aptkingston opened the pull request:
## Description
This PR adds an initial width setting to columns in tables. We had a width setting with the old table, but we didn't add it for the new table because
- You can drag to resize columns as you like
- The widths default to those in the data section, so should already be roughly correct

There are still valid use cases for having unique widths per table instance, so this PR restores that width setting. This setting is an "initial width" setting rather than an explicit width - that is to say that users can still resize columns as they like, but this setting will change the initial widths to those specified by the app builder.

![image](https://github.com/Budibase/budibase/assets/9075550/afab5af4-e258-452b-8e75-58856adda7f0)

## Launchcontrol
Add initial width setting to table columns.

aptkingston commented on packages/builder/src/components/design/settings/controls/DraggableList/DraggableList.svelte:
```diff
@@ -100,9 +100,6 @@
       }}
-      on:mousedown={() => {
-        get(store).actions.select()
-      }}
```
Not needed - this was just closing the previous popover on mouse down which isn't how we usually handle popovers so I'm removing this to keep things consistent with just clicks.

aptkingston commented on packages/builder/src/components/design/settings/controls/EditComponentPopover/index.js:
This custom position handler is no longer needed, as my updated logic for the usual position_dropdown util already handles this. I haven't specifically checked usage with the tours yet but I will.

deanhannigan replied:
I had a quick look at the tour stuff there and they are starting but just not visible

deanhannigan reviewed (changes requested):
Some minor issues but the changes are solid!

`Bug` - There seems to be some issues when moving GridBlock columns in the client. When attempting to move a column for the first time in the GridBlock is doesn't move. On the second attempt it moves back to the previous location.

`Bug` - Tours start but are no longer visible.

aptkingston commented:
> Some minor issues but the changes are solid!
> 
> `Bug` - Context menu for the Table now floating to the top in the builder preview <img alt="" width="450px" src="https://private-user-images.githubusercontent.com/5913006/331529432-5ce100ea-281b-449a-88d2-5a695972ee8f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5NTg1MjgsIm5iZiI6MTcxNTk1ODIyOCwicGF0aCI6Ii81OTEzMDA2LzMzMTUyOTQzMi01Y2UxMDBlYS0yODFiLTQ0OWEtODhkMi01YTY5NTk3MmVlOGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTdUMTUwMzQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGM2YjQ0MDJmY2FlNDA3ZmEwYjdmYmMxOTcxNTZiZmE2N2YzZmI5ZjRjOGQzNGRiMWI1NTVjNjMwNzBlNzU5NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aaN2rBkvKvscIOdI3SvSFdcB5WSRCB25Z_fqZHTC98c">

Fixed 👌 Was actually caused by a previous PR but I've addressed it here since it was easy enough.

deanhannigan reviewed (approved):
LGTM! The updates to the grid are solid! After a run through, I can see the context menu is back in place and no other issues remain 👍
